src/MoBi.Presentation/Presenter/Main/StatusBarPresenter.cs

```diff
@@ -118,14 +128,36 @@ public void Handle(ProgressingEvent eventToHandle)
             .WithValue(eventToHandle.ProgressPercent);
 
          update(StatusBarElements.ProgressStatus)
-            .WithCaption(eventToHandle.Message);
+            .WithCaption($"{activeSimulations} {eventToHandle.Message}");
       }
 
       public void Handle(ProgressDoneEvent eventToHandle)
       {
+         var message = eventToHandle is ProgressDoneWithMessageEvent ?
```

Comment:
why not listen to two messages? Handle specifically does it for you so you can be specific. There is not enough common code here and if there is, it can be in a function IMO